src/Common/src/System/Data/Common/FieldNameLookup.cs

```diff
@@ -6,16 +6,16 @@
 
 //------------------------------------------------------------------------------
 
-using System.Collections;
 using System.Data.Common;
 using System.Globalization;
+using System.Collections.Generic;
 
 namespace System.Data.ProviderBase
 {
     internal sealed class FieldNameLookup
     {
         // hashtable stores the index into the _fieldNames, match via case-sensitive
-        private Hashtable _fieldNameLookup;
+        private Dictionary<string, int> _fieldNameLookup;
```

Comment:
Nit: the above comment refers to "hashtable", which could be changed to "Dictionary"